nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/controller/StandardProcessorNode.java

```diff
@@ -1209,33 +1194,112 @@ protected Collection<ValidationResult> computeValidationErrors(final ValidationC
     }
 
     @Override
-    protected List<ValidationResult> validateConfig() {
+    public List<ValidationResult> validateConfig() {
 
         final List<ValidationResult> results = new ArrayList<>();
         final ParameterContext parameterContext = getParameterContext();
 
-        if (parameterContext == null && !this.parameterReferences.isEmpty()) {
+        if (parameterContext == null && !this.configurationParameterReferences.isEmpty()) {
             results.add(new ValidationResult.Builder()
-                    .subject(getName())
+                    .subject(RUN_SCHEDULE)
+                    .input("Parameter Context")
                     .valid(false)
                     .explanation("Processor configuration references one or more Parameters but no Parameter Context is currently set on the Process Group.")
                     .build());
         } else {
-            for (final ParameterReference paramRef : parameterReferences) {
-                if (!parameterContext.getParameter(paramRef.getParameterName()).isPresent() ) {
+            for (final ParameterReference paramRef : configurationParameterReferences) {
+                final Optional<Parameter> parameterRef = parameterContext.getParameter(paramRef.getParameterName());
+                if (!parameterRef.isPresent() ) {
                     results.add(new ValidationResult.Builder()
-                            .subject(getName())
+                            .subject(RUN_SCHEDULE)
+                            .input(paramRef.getParameterName())
                             .valid(false)
                             .explanation("Processor configuration references Parameter '" + paramRef.getParameterName() +
                                     "' but the currently selected Parameter Context does not have a Parameter with that name")
                             .build());
+                } else {
+                    final ParameterDescriptor parameterDescriptor = parameterRef.get().getDescriptor();
+                    if (parameterDescriptor.isSensitive()) {
+                        results.add(new ValidationResult.Builder()
+                                .subject(RUN_SCHEDULE)
+                                .input(parameterDescriptor.getName())
+                                .valid(false)
+                                .explanation("Processor configuration cannot reference sensitive parameters")
+                                .build());
+                    }
                 }
             }
-        }
 
+            final String schedulingPeriod = getSchedulingPeriod();
+            final String evaluatedSchedulingPeriod = evaluateParameter(schedulingPeriod);
+
+            if (evaluatedSchedulingPeriod != null) {
```

Comment:
This should never be `null`, should it?

Reply:
For User experience I would not like to check validation at first error found. As parameter substitution can be null if a parameter can not be substituted, to be able to continue validation checks, I need to check for null here. 
If the value is incorrect the Processor will be invalid and no null value will be set.